packages/ai/src/ai/common/validation.py

```
@@ -0,0 +1,135 @@
+        prompt: The raw prompt string.
+        max_tokens: The model's total token limit (context window).
+        token_counter: A callable that estimates token count for a string.
+
+    Returns:
+        The sanitized prompt string, ready for the API call.
+
+    Raises:
+        ValueError: If the prompt is empty or whitespace-only.
+    """
+    if not prompt or not prompt.strip():
+        raise ValueError('Prompt is empty or contains only whitespace.')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)